### PR TITLE
Feature/add capacity inputfield

### PR DESCRIPTION
**Pull request by Dskottke**

### CODE REVIEW

**Review by ZeshanShahid79 — APPROVED** [review]
add new line in itemtablerow and all fine

**Review by ZeshanShahid79 — APPROVED** [review]
wow great job

File: frontend/src/components/ItemTableRow.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            <td width="200px">{inEditMode.status && inEditMode.rowKey === item.id ? (
+                    <React.Fragment>
+                        <button
+                            className={"button-left"}
+                            onClick={() => updateItem()}
+                        >save
+                        </button>
+                        <button
+                            className={"button-right"}
+                            style={{marginLeft: 8}}
```
Comment (chris-yooo) [inline frontend/src/components/ItemTableRow.tsx:108]: why inlinestyle you got button right as class ?

File: frontend/src/components/ItemTableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                            onClick={() => onCancel()}
+                        >cancel
+                        </button>
+                    </React.Fragment>) :
+                <React.Fragment>
```
Comment (chris-yooo) [inline frontend/src/components/ItemTableRow.tsx:113]: <>  </> :D

File: frontend/src/components/ItemTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            })
+            .then(fetchData)
+    }
+    const handleInputCapacity = (event: ChangeEvent<HTMLInputElement>) => {
+        const validCapacity = event.target.value.replace(/\D/g, '')
+        setStorableValue(validCapacity)
+    }
+
+
+    const onCancel = () => {
```
Comment (chris-yooo) [inline frontend/src/components/ItemTableRow.tsx:72]: you can use an toggle here maybe :)

  const [hamburgerOpen, setHamburgerOpen] = useState(false);

  const toggleHamburger = () => {
    setHamburgerOpen(!hamburgerOpen);
  };

File: frontend/src/components/ItemTableRow.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            <td>{item.categoryName}</td>
+            <td>{item.issuingCountry}</td>
+            <td>{inEditMode.status && inEditMode.rowKey === item.id ? (
+                <input value={storableValue}
+                       onChange={handleInputCapacity}
+                />) : (
+                item.storableValue)}
+            </td>
+            <td width="200px">{inEditMode.status && inEditMode.rowKey === item.id ? (
+                    <React.Fragment>
```
Comment (chris-yooo) [inline frontend/src/components/ItemTableRow.tsx:100]: fragemennnnnnnnnnt :D  <>

File: frontend/src/components/ItemNavigation.tsx [hunk 5]
```diff
@@ -44,23 +51,35 @@ function ItemNavigation(props: itemNavigationProbs) {
                     props.setErrorModal(true);
                     props.setErrorMessage("found no matching item with ean")
                 }
+                if (error.response.status === 412) {
```
Comment (chris-yooo) [inline frontend/src/components/ItemNavigation.tsx:54]: nice

**Review by chris-yooo — CHANGES_REQUESTED** [review]
:p

**Comment by chris-yooo** [comment]
you resolved the smells, but you want to change some empty lines and other things that your code looks fine ? write me to approve your code :)

File: frontend/src/components/ItemPage.tsx [hunk 6]
```diff
@@ -21,6 +21,7 @@ function ItemPage() {
     const fetchData = () => {
         axios.get("/api/items/")
             .then(response => {
+                console.log(response.data)
```
Comment (chris-yooo) [inline frontend/src/components/ItemPage.tsx:24]: you dont need console.log in prod code :S

File: backend/src/test/java/capstone/storage/backend/ItemServiceTest.java [hunk 7]
```diff
@@ -103,54 +111,93 @@ void deleteItemById() {
...
+    @DisplayName("method : isNullOrEmpty -> should return false when all AddItemDto fields are not empty")
+    void checkIfIsNullOrEmptyReturnsFalseWhenAddITemDtoFieldsAreNotEmptyString() {
+        //GIVEN
+        AddItemDto addItemDto = new AddItemDto("123", "1", "1");
+        //WHEn
+        boolean actual = itemService.isNullOrEmpty(addItemDto);
         boolean expected = false;
         //THEN
-        assertEquals(actual, expected);
+        assertEquals(expected, actual);
```
Comment (chris-yooo) [inline backend/src/test/java/capstone/storage/backend/ItemServiceTest.java:187]: :D

File: frontend/src/components/ItemTableRow.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+            <td>{item.categoryName}</td>
+            <td>{item.issuingCountry}</td>
+            <td>{onEditOpen ? (
+                <input value={storableValue}
+                       onChange={handleInputCapacity}
+                />) : (
+                item.storableValue)}
+            </td>
+            <td width="200px">{onEditOpen ? (
+                    <>
```
Comment (chris-yooo) [inline frontend/src/components/ItemTableRow.tsx:83]: <3

File: frontend/src/components/ItemTableRow.tsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                            onClick={() => updateItem()}
+                        >save
+                        </button>
+                        <button
+                            className={"button-right"}
+                            onClick={() => setOnEditOpen(!onEditOpen)}
+                        >cancel
+                        </button>
+                    </>) :
+                <>
```
Comment (chris-yooo) [inline frontend/src/components/ItemTableRow.tsx:95]: <3

File: backend/src/main/java/capstone/storage/backend/ItemControllerAdvice.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @ExceptionHandler(ItemResponseEanNullException.class)
+    public ResponseEntity<String> handleEanApiResponseException(ItemResponseEanNullException ex) {
+
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                .body(ex.getMessage());
+    }
+
+}
```
Comment (chris-yooo) [inline backend/src/main/java/capstone/storage/backend/ItemControllerAdvice.java:56]: nice when you get time you can explain that 4 me :D

**Review by chris-yooo — APPROVED** [review]
i think you know what you do so i took an approve with the 1 little thing that you have to change -> the console log. So nice work

File: backend/src/main/java/capstone/storage/backend/ItemControllerAdvice.java (hunk 10 above)
Comment (Dskottke) [reply backend/src/main/java/capstone/storage/backend/ItemControllerAdvice.java:56]: =D of course! thank u !

File: frontend/src/components/ItemPage.tsx (hunk 6 above)
Comment (Dskottke) [reply frontend/src/components/ItemPage.tsx:24]: thank u :D haha i forgot that <3

**Review by chris-yooo — APPROVED** [review]

**Pull request merged**